proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
     -> UnsafeMutablePointer<T>
 
-  func initialize<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func initializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
   // The `move` APIs deinitialize the memory at `from`.
-  func moveInitialize<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func moveInitializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
-  func load<T>(_: T.Type) -> T
-  func load<T>(_: T.Type, atByteOffset: Int) -> T
-  func load<T>(fromContiguous: T.Type, atIndex: Int) -> T
+  func load<T>(as: T.Type) -> T
+  func load<T>(fromByteOffset: Int, as: T.Type) -> T
+  func load<T>(fromContiguous: T.Type, at: Int) -> T
```

Comment:
I’m not convinced this API is a good idea.  Is there really a strong enough use-case for it?  

My theory: if you are dealing with actual contiguous `T`’s you should be using typed access.  Otherwise, your `T` is unlikely to line up on a multiple of `MemoryLayout<T>.stride` anyhow.  What’s left over?  Are the uses important enough to complicate the API for?